src/wazuh_modules/task_manager/wm_task_manager_commands.c

```diff
@@ -239,4 +253,62 @@ void* wm_task_manager_clean_tasks(void *arg) {
     return NULL;
 }
 
+STATIC cJSON* wm_task_manager_send_message_to_wdb(const char *command, cJSON *parameters, int *error_code) {
+    cJSON *root = cJSON_CreateObject();
+    cJSON *response = cJSON_CreateObject();
+
+    const char *json_err;
+    int result = 0;
+    char *parameters_in_str = NULL;
+    char wdbquery[WDBQUERY_SIZE] = "";
+    char wdboutput[WDBOUTPUT_SIZE] = "";
+    char *payload = NULL;
+    int socket = -1;
+
+
+    parameters_in_str = cJSON_PrintUnformatted(parameters);
+    snprintf(wdbquery, sizeof(wdbquery), "task %s %s", command, parameters_in_str);
+    os_free(parameters_in_str);
+
+    result = wdbc_query_ex(&socket, wdbquery, wdboutput, sizeof(wdboutput));
+    wdbc_close(&socket);
+
+    switch (result) {
+        case OS_SUCCESS:
+            if (WDBC_OK == wdbc_parse_result(wdboutput, &payload)) {
+                // Parsing payload
+                if (response = cJSON_ParseWithOpts(payload, &json_err, 0), !response) {
+                    mterror(WM_TASK_MANAGER_LOGTAG, MOD_TASK_PARSE_JSON_ERROR, payload);
+                    *error_code = WM_TASK_PARSE_ERROR;
+                    cJSON_Delete(response);
+                    cJSON_Delete(root);
+                }
+
+                cJSON_AddStringToObject(root,"output",wdboutput);
+                cJSON_AddItemToObject(root,"payload",response);
+            }
+            else {
+                mdebug1("Tasks DB Error reported in the result of the query");
+                cJSON_AddStringToObject(root,"output",wdboutput);
+                cJSON_AddItemToObject(root,"payload",payload);
+            }
+            break;
+        case OS_INVALID:
```

Comment:
I think this case is not necessary. Use only default and remember to add a break at the end.